libs/beacon/src/beacon_service.cpp

```diff
@@ -252,6 +252,43 @@ void BeaconService::StartNewCabinet(CabinetMemberList members, uint32_t threshol
   cabinet_creator_.QueueSetup(beacon);
 }
 
+void BeaconService::StartNewCabinet(CabinetMemberList members, uint32_t threshold,
```

Comment:
There is quite a lot of duplicated code between the two `StartNewCabinet` methods. Is it worth doing a little more refactoring so that this can be reused?

Reply:
Yes, this did occur to me as well. I will change this.